pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1102,9 +1088,13 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(clusterName string, service *v1.
 	var monitorIDs []string
 	for _, listener := range listenerList {
 		pool, err := getPoolByListenerID(lbaas.network, loadbalancer.ID, listener.ID)
-		if err != nil {
+		if err != nil && err != ErrNotFound {
 			return fmt.Errorf("Error getting pool for listener %s: %v", listener.ID, err)
 		}
+		if err == ErrNotFound {
+			// Unknown error, retry later
+			continue
```

Comment:
Will this function be retried if it does *not* return an error?  It's been a while since I read the calling code, but I thought we had to return an error in order to be retried ...